.github/workflows/parse_and_plot.yml

```diff
@@ -0,0 +1,89 @@
+# Syntax reference:
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions
+
+name: Parse and plot
+permissions: read-all
+
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+      - 'LICENSE'
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  DEBIAN_FRONTEND: noninteractive
+
+jobs:
+  parse-and-plot:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Update packages
+        run: sudo apt-get update
+
+      - name: Install tornettools dependencies
+        run: sudo apt-get install -y
+          python3
+          python3-dev
+          python3-pip
+          libxml2
+          libxml2-dev
+          libxslt1.1
+          libxslt1-dev
+          libpng16-16
+          libpng-dev
+          libfreetype6
+          libfreetype6-dev
+          libblas-dev
+          liblapack-dev
+
+      - name: Build tornettools
+        run: |
+          pip3 install wheel
+          pip3 install -r requirements.txt
+          pip3 install -I .
+
+      - name: Copy golden parsed logs
+        run: |
+          mkdir golden_reparsed
+          cp .github/workflows/golden_tornet/*.json.xz golden_reparsed
+          cp .github/workflows/golden_tornet/tornettools.generate.*.log* golden_reparsed
+
+      - name: Re-parse
+        # -c must match the convergence time used to generate the golden data,
+        # in run_all_steps.yml
+        #
+        # We use --skip-raw to skip trying to parse the raw logs again, so that
+        # we don't need to check the raw logs into version control, and since
+        # the tgen and oniontrace parsing are done by those respective tools,
+        # which we're not trying to test here.
+        run: tornettools parse -c 300 --skip-raw golden_reparsed
+
+      # tornet.plot.data files should usually match exactly.
+      # If a difference found in this step is expected, then the PR generating
+      # the diff should also update the golden output (generally by using the
+      # artifact from run_all_steps).
+      - name: Diff
+        run: for f in .github/workflows/golden_tornet/tornet.plot.data/*.json;
```

Comment:
I don't really understand the purpose of this step. Isn't the purpose of most PRs that change the parsing/plotting code to change the tornettools output in some observable way? Why do we want to check that all of the generated files are exactly the same?

Also, I think the simulation data contains information on the real time? Won't that cause this step to fail since it's non-deterministic? Or is this information stored in files that don't end in `.json`? If so, maybe it would be better if this step only compares explicit files rather than a glob, so that we can be explicit about which data is supposed to be deterministic.

Reply:
Still confused about this, but I left a comment above.